### PR TITLE
Add: toolchain to build packages, preview website locally, and a bonus section

### LINKED ISSUE #15: Rtools not found
When running `devtools::check()`, things start well, but then there's an error about Rtools:

```
══ Building ══════════════════════════════════════════════════════════════════
Setting env vars:
• CFLAGS    : -Wall -pedantic
• CXXFLAGS  : -Wall -pedantic
• CXX11FLAGS: -Wall -pedantic
• CXX14FLAGS: -Wall -pedantic
• CXX17FLAGS: -Wall -pedantic
• CXX20FLAGS: -Wall -pedantic
── R CMD build ───────────────────────────────────────────────────────────────
✔  checking for file 'C:\Users\dalonsoa\Projects\csvyr/DESCRIPTION' (960ms)
─  preparing 'csvyr': (933ms)
✔  checking DESCRIPTION meta-information
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
   Omitted 'LazyData' from DESCRIPTION
─  building 'csvyr_0.1.0.tar.gz'
   
══ Checking ══════════════════════════════════════════════════════════════════
Setting env vars:
• _R_CHECK_CRAN_INCOMING_REMOTE_               : FALSE
• _R_CHECK_CRAN_INCOMING_                      : FALSE
• _R_CHECK_FORCE_SUGGESTS_                     : FALSE
• _R_CHECK_PACKAGES_USED_IGNORE_UNUSED_IMPORTS_: FALSE
• NOT_CRAN                                     : true
trying URL 'https://cran.rstudio.com/bin/windows/Rtools/rtools44/files/rtools44-6104-6039.exe'
Content type 'application/x-msdownload' length 449027200 bytes (428.2 MB)
downloaded 428.2 MB

Error: Could not find tools necessary to compile a package
Call `pkgbuild::check_build_tools(debug = TRUE)` to diagnose the problem.
```
Then, after a while, it automatically launches the installation of `Rtools` after asking for user's approval. Installation takes a while but seems to conclude without any issues. Then, I re-run `devtools::check()` that ends up with the License note, as expected:

```
── R CMD check results ────────────────────────────────────── csvyr 0.1.0 ────
Duration: 19.4s

❯ checking DESCRIPTION meta-information ... NOTE
  Non-standard licence specification:
    What license is it under?
  Standardisable: FALSE

0 errors ✔ | 0 warnings ✔ | 1 note ✖
```

I had a fresh installation of R, so probably this is not really an issue for someone who has used R already, but it might be worrying for newbies. I'd suggest an explanation that something like this might happen.

### PR DESCRIPTION
**Pull request by SaranjeetKaur**
Fixes #15 and #18 

Added instructions for setting up the development environment for building R packages on Windows, macOS, and Linux. This includes installing `Rtools`, `Xcode`, and `r-base-dev` respectively.

Additionally,

* Added a GitHub Actions workflow for linting the R package using `lintr`.
* Added a GitHub Actions workflow for rendering Rmarkdown files.
* Added a bonus section on adding GitHub actions, customizing the package website, and adding a package to `DESCRIPTION`. 
* Added an R CMD-check badge to the `README.md`.
* Removed a redundant image.
* Added steps to preview site locally.

### CODE REVIEW

**Comment by SaranjeetKaur** [comment]
@dalonsoa - please review these changes. (Not sure why I am not able to add you as a reviewer here!)

File: README.Rmd [hunk 1]
```diff
@@ -28,6 +20,14 @@ The goal of repositoryr is to provide a step by step guide to creating a R packa
   - Use [Posit Cloud](https://posit.cloud/) (online) OR
   - Install it via Chocolatey (first install R using `choco install r.project` and then RStudio using `choco install r.studio`) OR 
   - Install it via Homebrew.
+  
+## Toolchain for building R packages
```
Comment (dalonsoa) [inline README.Rmd:24]: Nice!

File: README.Rmd [hunk 2]
```diff
@@ -103,20 +109,42 @@ git push --set-upstream origin main
...
 - To configure the package to use and deploy pkgdown, run `usethis::use_pkgdown_github_pages()`.
-- To preview your site locally before publishing, run `pkgdown::build_site()`.
 - This will add the necessary components and sets up GitHub Actions for automatic site building when deploying. 
+- To preview your site locally before publishing, run `pkgdown::build_site()`. If you see a warning that the "package is in use and will not be installed", then:
+
+  - Run `search()` to see if the package is loaded.
+  - If it is, run `detach("package:pkgname", unload = TRUE)` to unload it.
+  - Run `search()` again to verify that the package is no longer loaded.
+  - Then run `pkgdown::build_site()` again to preview the site locally.
+
```
Comment (dalonsoa) [inline README.Rmd:123]: Good! This works 👍

**Review by dalonsoa — APPROVED** [review]
Very nice.

**Pull request merged**